client/src/components/events-components/EventCard.tsx

```diff
@@ -7,11 +7,12 @@ import { deleteEventRsvp, eventRsvp } from "@/utils/eventService";
 
 interface EventProps {
   event: EventWithRsvp;
```

Comment:
```suggestion
  event: Event;
  isRSVPed: boolean;
```

I would structure the props differently, because they are for different concerns:
- `event` to render the Event details
- `rsvp` for controlling the RSVP button look